### ISSUE TITLE
BTD crashing with invalid particle error

### ISSUE THREAD

**Issue by RevathiJambunathan** [ECP-WarpX/WarpX#3283]
For a 2D psatd test case, the plotfile format for BTD crashed with 
```
amrex::Abort::0::ParticleContainer::locateParticle(): invalid particle. !!!
SIGABRT
See Backtrace.0.3 file for details
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI COMMUNICATOR 3 DUP FROM 0
with errorcode 6.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
```

Looking into this further, the particle in question is located at z=1.72299362e-05, and it gets mapped to index `(17,1536) `
However, the particle BoxArray is 
```
 ba : (BoxArray maxbox(1)
       m_ref->m_hash_sig(0)
       ((0,1280) (63,1535) (0,0)) )
```
Thus the particle is out-of-bounds. 
The input file was provided by @RTSandberg

**Comment by RTSandberg** [comment]
Running on Perlmutter in debug mode, the simulation crashes with the following message:
```
STEP 1 ends. TIME = 4.365974052e-16 DT = 4.365974052e-16 
Evolve time = 0.018899698 s; This step = 0.018899698 s; Avg. per step = 0.018899698 s

STEP 2 starts ...
 (64,559,0,0) is out of bound (0:63,512:639,0:0,0:0)
... <lots of similar out of bound messages> ...
 (-1,2116,0,0) is out of bound (0:63,2048:2175,0:0,0:0) 
```
The `WarpX.e` file says
```
/global/u2/r/rsand/src/WarpX/build_perlmutter_lattice/_deps/fetchedamrex-src/Src/Base/AMReX.H:152: void amrex::Abort(const char *): block: [26,0,0], thread: [75,0,0] Assertion `0` failed.
...
/global/u2/r/rsand/src/WarpX/build_perlmutter_lattice/_deps/fetchedamrex-src/Src/Base/AMReX.H:152: void amrex::Abort(const char *): block: [16,0,0], thread: [62,0,0] Assertion `0` failed.
amrex::Abort::0::CUDA error 710 in file /global/homes/r/rsand/src/WarpX/build_perlmutter_lattice/_deps/fetchedamrex-src/Src/Base/AMReX_GpuDevice.cpp line 660: device-side assert triggered !!!
SIGABRT
```

The input deck is here:
[inputs.txt](https://github.com/ECP-WarpX/WarpX/files/9253965/inputs.txt)

**Comment by RemiLehe** [comment]
@RevathiJambunathan Is this issue now fixed, thanks to #3288 and #3285?
If yes, do you want to close the issue?

**Comment by RevathiJambunathan** [comment]
Yes. This is resolved.
